From 1e79978d4004a5a18e69a0e187c97f3565e86759 Mon Sep 17 00:00:00 2001
From: Chanh Nguyen <chanh@os.amperecomputing.com>
Date: Fri, 23 Aug 2024 04:41:02 +0000
Subject: [PATCH] Redfish: Add message registries for Ampere events

Add Ampere defined registries for events that have not defined by
OpenBMC.

Also add the message registry for the power button long press.

Signed-off-by: Hieu Huynh <hieuh@os.amperecomputing.com>
Signed-off-by: Chanh Nguyen <chanh@os.amperecomputing.com>
---
 redfish-core/include/registries/openbmc.json  |  71 +++
 .../registries/openbmc_message_registry.hpp   | 490 +++++++++++-------
 2 files changed, 371 insertions(+), 190 deletions(-)

diff --git a/redfish-core/include/registries/openbmc.json b/redfish-core/include/registries/openbmc.json
index 885fef46..74d3dee2 100644
--- a/redfish-core/include/registries/openbmc.json
+++ b/redfish-core/include/registries/openbmc.json
@@ -14,6 +14,33 @@
             "Resolution": "None.",
             "Severity": "Warning"
         },
+        "AmpereCritical": {
+            "Description": "Indicates an Ampere critical event",
+            "Message": "%1: %2",
+            "MessageSeverity": "Critical",
+            "NumberOfArgs": 2,
+            "ParamTypes": ["string", "string"],
+            "Resolution": "None.",
+            "Severity": "Critical"
+        },
+        "AmpereEvent": {
+            "Description": "Indicates an Ampere OK event",
+            "Message": "%1",
+            "MessageSeverity": "OK",
+            "NumberOfArgs": 1,
+            "ParamTypes": ["string"],
+            "Resolution": "None.",
+            "Severity": "OK"
+        },
+        "AmpereWarning": {
+            "Description": "Indicates an Ampere warning event",
+            "Message": "%1: %2",
+            "MessageSeverity": "Warning",
+            "NumberOfArgs": 2,
+            "ParamTypes": ["string", "string"],
+            "Resolution": "None.",
+            "Severity": "Warning"
+        },
         "AtScaleDebugConnected": {
             "Description": "Indicates At-Scale Debug connection has been established",
             "Message": "At-Scale Debug service is now connected %1",
@@ -821,6 +848,24 @@
             "Resolution": "None.",
             "Severity": "Critical"
         },
+        "MemoryExtendedECCCEData": {
+            "Description": "Indicates additional Memory ECC Correctable error data.",
+            "Message": "Memory ECC Correctable Error Extended Data. bank=%1 row=%2 column=%3.",
+            "MessageSeverity": "Critical",
+            "NumberOfArgs": 3,
+            "ParamTypes": ["number", "number", "number"],
+            "Resolution": "None.",
+            "Severity": "Critical"
+        },
+        "MemoryExtendedECCUEData": {
+            "Description": "Indicates additional Memory ECC Uncorrectable error data.",
+            "Message": "Memory ECC Uncorrectable Error Extended Data. bank=%1 row=%2 column=%3.",
+            "MessageSeverity": "Critical",
+            "NumberOfArgs": 3,
+            "ParamTypes": ["number", "number", "number"],
+            "Resolution": "None.",
+            "Severity": "Critical"
+        },
         "MemoryParityCommandAndAddress": {
             "Description": "Indicates a Command and Address parity error.",
             "Message": "Command and Address parity error. Socket=%1 Channel=%2 DIMM=%3 ChannelValid=%4 DIMMValid=%5.",
@@ -1187,6 +1232,14 @@
             "Resolution": "None.",
             "Severity": "OK"
         },
+        "PowerButtonLongPressed": {
+            "Description": "Indicates that the power button was long pressed.",
+            "Message": "Power Button Long Pressed.",
+            "MessageSeverity": "OK",
+            "NumberOfArgs": 0,
+            "Resolution": "None.",
+            "Severity": "OK"
+        },
         "PowerRestorePolicyApplied": {
             "Description": "Indicates that power was restored and the BMC has applied the restore policy.",
             "Message": "Power restore policy applied.",
@@ -1654,6 +1707,24 @@
             "Resolution": "None.",
             "Severity": "Critical"
         },
+        "TotalPowerConsumptionDropBelowTheLimit": {
+            "Description": "Indicate the total power consumption drops below the power limit",
+            "Message": "Total power consumption drops below power limit. The total power consumption is %1, the power limit is %2.",
+            "MessageSeverity": "OK",
+            "NumberOfArgs": 2,
+            "ParamTypes": ["number", "number"],
+            "Resolution": "None.",
+            "Severity": "OK"
+        },
+        "TotalPowerConsumptionExceedTheLimit": {
+            "Description": "Indicate the total power consumption exceeds the power limit",
+            "Message": "Total power consumption exeeds power limit. The total power consumption is %1, the power limit is %2.",
+            "MessageSeverity": "Critical",
+            "NumberOfArgs": 2,
+            "ParamTypes": ["number", "number"],
+            "Resolution": "None.",
+            "Severity": "Critical"
+        },
         "VoltageRegulatorOverheated": {
             "Description": "Indicates that the specified voltage regulator overheated.",
             "Message": "%1 Voltage Regulator Overheated.",
diff --git a/redfish-core/include/registries/openbmc_message_registry.hpp b/redfish-core/include/registries/openbmc_message_registry.hpp
index 35625042..5ca92651 100644
--- a/redfish-core/include/registries/openbmc_message_registry.hpp
+++ b/redfish-core/include/registries/openbmc_message_registry.hpp
@@ -48,6 +48,44 @@ constexpr std::array registry =
             },
             "None.",
         }},
+    MessageEntry{
+        "AmpereCritical",
+        {
+            "Indicates an Ampere critical event",
+            "%1: %2",
+            "Critical",
+            2,
+            {
+                "string",
+                "string",
+            },
+            "None.",
+        }},
+    MessageEntry{
+        "AmpereEvent",
+        {
+            "Indicates an Ampere OK event",
+            "%1",
+            "OK",
+            1,
+            {
+                "string",
+            },
+            "None.",
+        }},
+    MessageEntry{
+        "AmpereWarning",
+        {
+            "Indicates an Ampere warning event",
+            "%1: %2",
+            "Warning",
+            2,
+            {
+                "string",
+                "string",
+            },
+            "None.",
+        }},
     MessageEntry{
         "AtScaleDebugConnected",
         {
@@ -1132,6 +1170,34 @@ constexpr std::array registry =
             },
             "None.",
         }},
+    MessageEntry{
+        "MemoryExtendedECCCEData",
+        {
+            "Indicates additional Memory ECC Correctable error data.",
+            "Memory ECC Correctable Error Extended Data. bank=%1 row=%2 column=%3.",
+            "Critical",
+            3,
+            {
+                "number",
+                "number",
+                "number",
+            },
+            "None.",
+        }},
+    MessageEntry{
+        "MemoryExtendedECCUEData",
+        {
+            "Indicates additional Memory ECC Uncorrectable error data.",
+            "Memory ECC Uncorrectable Error Extended Data. bank=%1 row=%2 column=%3.",
+            "Critical",
+            3,
+            {
+                "number",
+                "number",
+                "number",
+            },
+            "None.",
+        }},
     MessageEntry{
         "MemoryParityCommandAndAddress",
         {
@@ -1696,6 +1762,16 @@ constexpr std::array registry =
             {},
             "None.",
         }},
+    MessageEntry{
+        "PowerButtonLongPressed",
+        {
+            "Indicates that the power button was long pressed.",
+            "Power Button Long Pressed.",
+            "OK",
+            0,
+            {},
+            "None.",
+        }},
     MessageEntry{
         "PowerRestorePolicyApplied",
         {
@@ -2326,6 +2402,32 @@ constexpr std::array registry =
             {},
             "None.",
         }},
+    MessageEntry{
+        "TotalPowerConsumptionDropBelowTheLimit",
+        {
+            "Indicate the total power consumption drops below the power limit",
+            "Total power consumption drops below power limit. The total power consumption is %1, the power limit is %2.",
+            "OK",
+            2,
+            {
+                "number",
+                "number",
+            },
+            "None.",
+        }},
+    MessageEntry{
+        "TotalPowerConsumptionExceedTheLimit",
+        {
+            "Indicate the total power consumption exceeds the power limit",
+            "Total power consumption exeeds power limit. The total power consumption is %1, the power limit is %2.",
+            "Critical",
+            2,
+            {
+                "number",
+                "number",
+            },
+            "None.",
+        }},
     MessageEntry{
         "VoltageRegulatorOverheated",
         {
@@ -2344,195 +2446,203 @@ constexpr std::array registry =
 enum class Index
 {
     aDDDCCorrectable = 0,
-    atScaleDebugConnected = 1,
-    atScaleDebugConnectionFailed = 2,
-    atScaleDebugDisconnected = 3,
-    atScaleDebugFeatureDisabled = 4,
-    atScaleDebugFeatureDisabledAtHardware = 5,
-    atScaleDebugFeatureEnabled = 6,
-    atScaleDebugFeatureEnabledAtHardware = 7,
-    atScaleDebugSpecialUserDisabled = 8,
-    atScaleDebugSpecialUserEnabled = 9,
-    bIOSAttributesChanged = 10,
-    bIOSBoot = 11,
-    bIOSFirmwarePanicReason = 12,
-    bIOSFirmwareRecoveryReason = 13,
-    bIOSFirmwareResiliencyError = 14,
-    bIOSPOSTCode = 15,
-    bIOSPOSTError = 16,
-    bIOSRecoveryComplete = 17,
-    bIOSRecoveryStart = 18,
-    bMCBootReason = 19,
-    bMCFirmwarePanicReason = 20,
-    bMCFirmwareRecoveryReason = 21,
-    bMCFirmwareResiliencyError = 22,
-    bMCKernelPanic = 23,
-    cPLDFirmwarePanicReason = 24,
-    cPLDFirmwareRecoveryReason = 25,
-    cPLDFirmwareResiliencyError = 26,
-    cPUError = 27,
-    cPUMismatch = 28,
-    cPUThermalTrip = 29,
-    chassisIntrusionDetected = 30,
-    chassisIntrusionReset = 31,
-    componentOverTemperature = 32,
-    dCPowerOff = 33,
-    dCPowerOn = 34,
-    driveError = 35,
-    eventLogCleared = 36,
-    fanInserted = 37,
-    fanRedundancyLost = 38,
-    fanRedundancyRegained = 39,
-    fanRemoved = 40,
-    firmwareActivationCompleted = 41,
-    firmwareActivationFailed = 42,
-    firmwareActivationStarted = 43,
-    firmwareResiliencyError = 44,
-    firmwareUpdateCompleted = 45,
-    firmwareUpdateFailed = 46,
-    firmwareUpdateStaged = 47,
-    firmwareUpdateStarted = 48,
-    generalFirmwareSecurityViolation = 49,
-    iPMIWatchdog = 50,
-    intelUPILinkWidthReducedToHalf = 51,
-    intelUPILinkWidthReducedToQuarter = 52,
-    invalidLoginAttempted = 53,
-    invalidUpload = 54,
-    inventoryAdded = 55,
-    inventoryRemoved = 56,
-    lanLost = 57,
-    lanRegained = 58,
-    legacyPCIPERR = 59,
-    legacyPCISERR = 60,
-    mEAutoConfigFailed = 61,
-    mEAutoConfigSuccess = 62,
-    mEBootGuardHealthEvent = 63,
-    mECpuDebugCapabilityDisabled = 64,
-    mEDirectFlashUpdateRequested = 65,
-    mEExceptionDuringShutdown = 66,
-    mEFactoryResetError = 67,
-    mEFactoryRestore = 68,
-    mEFirmwareException = 69,
-    mEFirmwarePanicReason = 70,
-    mEFirmwareRecoveryReason = 71,
-    mEFirmwareResiliencyError = 72,
-    mEFlashEraseError = 73,
-    mEFlashStateInformation = 74,
-    mEFlashStateInformationWritingEnabled = 75,
-    mEFlashVerificationError = 76,
-    mEFlashWearOutWarning = 77,
-    mEImageExecutionFailed = 78,
-    mEInternalError = 79,
-    mEManufacturingError = 80,
-    mEMctpInterfaceError = 81,
-    mEMultiPchModeMisconfig = 82,
-    mEPeciOverDmiError = 83,
-    mEPttHealthEvent = 84,
-    mERecoveryGpioForced = 85,
-    mERestrictedMode = 86,
-    mESmbusLinkFailure = 87,
-    mEUmaError = 88,
-    mEUnsupportedFeature = 89,
-    manufacturingModeEntered = 90,
-    manufacturingModeExited = 91,
-    memoryECCCorrectable = 92,
-    memoryECCUncorrectable = 93,
-    memoryParityCommandAndAddress = 94,
-    memoryParityNotKnown = 95,
-    memoryRASConfigurationDisabled = 96,
-    memoryRASConfigurationEnabled = 97,
-    memoryRASModeDisabled = 98,
-    memoryRASModeEnabled = 99,
-    memoryThermTrip = 100,
-    mirroringRedundancyDegraded = 101,
-    mirroringRedundancyFull = 102,
-    nMIButtonPressed = 103,
-    nMIDiagnosticInterrupt = 104,
-    pCIeCorrectableAdvisoryNonFatal = 105,
-    pCIeCorrectableBadDLLP = 106,
-    pCIeCorrectableBadTLP = 107,
-    pCIeCorrectableHeaderLogOverflow = 108,
-    pCIeCorrectableInternal = 109,
-    pCIeCorrectableLinkBWChanged = 110,
-    pCIeCorrectableReceiverError = 111,
-    pCIeCorrectableReplayNumRollover = 112,
-    pCIeCorrectableReplayTimerTimeout = 113,
-    pCIeCorrectableUnspecifiedAERError = 114,
-    pCIeFatalACSViolation = 115,
-    pCIeFatalAtomicEgressBlocked = 116,
-    pCIeFatalCompleterAbort = 117,
-    pCIeFatalCompletionTimeout = 118,
-    pCIeFatalDataLinkLayerProtocol = 119,
-    pCIeFatalECRCError = 120,
-    pCIeFatalFlowControlProtocol = 121,
-    pCIeFatalMCBlockedTLP = 122,
-    pCIeFatalMalformedTLP = 123,
-    pCIeFatalPoisonedTLP = 124,
-    pCIeFatalReceivedErrNonFatalMessage = 125,
-    pCIeFatalReceivedFatalMessageFromDownstream = 126,
-    pCIeFatalReceiverBufferOverflow = 127,
-    pCIeFatalSurpriseLinkDown = 128,
-    pCIeFatalTLPPrefixBlocked = 129,
-    pCIeFatalUncorrectableInternal = 130,
-    pCIeFatalUnexpectedCompletion = 131,
-    pCIeFatalUnspecifiedNonAERFatalError = 132,
-    pCIeFatalUnsupportedRequest = 133,
-    powerButtonPressed = 134,
-    powerRestorePolicyApplied = 135,
-    powerSupplyConfigurationError = 136,
-    powerSupplyConfigurationErrorRecovered = 137,
-    powerSupplyFailed = 138,
-    powerSupplyFailurePredicted = 139,
-    powerSupplyFanFailed = 140,
-    powerSupplyFanRecovered = 141,
-    powerSupplyInserted = 142,
-    powerSupplyPowerGoodFailed = 143,
-    powerSupplyPowerLost = 144,
-    powerSupplyPowerRestored = 145,
-    powerSupplyPredictedFailureRecovered = 146,
-    powerSupplyRecovered = 147,
-    powerSupplyRemoved = 148,
-    powerUnitDegradedFromNonRedundant = 149,
-    powerUnitDegradedFromRedundant = 150,
-    powerUnitNonRedundantFromInsufficient = 151,
-    powerUnitNonRedundantInsufficient = 152,
-    powerUnitNonRedundantSufficient = 153,
-    powerUnitRedundancyDegraded = 154,
-    powerUnitRedundancyLost = 155,
-    powerUnitRedundancyRegained = 156,
-    resetButtonPressed = 157,
-    sELEntryAdded = 158,
-    securityBoot2ndFlashEnabled = 159,
-    securityP2aBridgeEnabled = 160,
-    securityUartPortDebugEnabled = 161,
-    securityUserNonRootUidZeroAssigned = 162,
-    securityUserNonRootUidZeroRemoved = 163,
-    securityUserRootDisabled = 164,
-    securityUserRootEnabled = 165,
-    securityUserStrongHashAlgoRestored = 166,
-    securityUserUnsupportedShellEnabled = 167,
-    securityUserUnsupportedShellRemoved = 168,
-    securityUserWeakHashAlgoEnabled = 169,
-    sensorThresholdCriticalHighGoingHigh = 170,
-    sensorThresholdCriticalHighGoingLow = 171,
-    sensorThresholdCriticalLowGoingHigh = 172,
-    sensorThresholdCriticalLowGoingLow = 173,
-    sensorThresholdWarningHighGoingHigh = 174,
-    sensorThresholdWarningHighGoingLow = 175,
-    sensorThresholdWarningLowGoingHigh = 176,
-    sensorThresholdWarningLowGoingLow = 177,
-    serviceFailure = 178,
-    serviceStarted = 179,
-    sparingRedundancyDegraded = 180,
-    sparingRedundancyFull = 181,
-    ssbThermalTrip = 182,
-    systemInterfaceDisabledProvisioned = 183,
-    systemInterfaceUnprovisioned = 184,
-    systemInterfaceWhitelistProvisioned = 185,
-    systemPowerGoodFailed = 186,
-    systemPowerLost = 187,
-    systemPowerOffFailed = 188,
-    systemPowerOnFailed = 189,
-    voltageRegulatorOverheated = 190,
+    ampereCritical = 1,
+    ampereEvent = 2,
+    ampereWarning = 3,
+    atScaleDebugConnected = 4,
+    atScaleDebugConnectionFailed = 5,
+    atScaleDebugDisconnected = 6,
+    atScaleDebugFeatureDisabled = 7,
+    atScaleDebugFeatureDisabledAtHardware = 8,
+    atScaleDebugFeatureEnabled = 9,
+    atScaleDebugFeatureEnabledAtHardware = 10,
+    atScaleDebugSpecialUserDisabled = 11,
+    atScaleDebugSpecialUserEnabled = 12,
+    bIOSAttributesChanged = 13,
+    bIOSBoot = 14,
+    bIOSFirmwarePanicReason = 15,
+    bIOSFirmwareRecoveryReason = 16,
+    bIOSFirmwareResiliencyError = 17,
+    bIOSPOSTCode = 18,
+    bIOSPOSTError = 19,
+    bIOSRecoveryComplete = 20,
+    bIOSRecoveryStart = 21,
+    bMCBootReason = 22,
+    bMCFirmwarePanicReason = 23,
+    bMCFirmwareRecoveryReason = 24,
+    bMCFirmwareResiliencyError = 25,
+    bMCKernelPanic = 26,
+    cPLDFirmwarePanicReason = 27,
+    cPLDFirmwareRecoveryReason = 28,
+    cPLDFirmwareResiliencyError = 29,
+    cPUError = 30,
+    cPUMismatch = 31,
+    cPUThermalTrip = 32,
+    chassisIntrusionDetected = 33,
+    chassisIntrusionReset = 34,
+    componentOverTemperature = 35,
+    dCPowerOff = 36,
+    dCPowerOn = 37,
+    driveError = 38,
+    eventLogCleared = 39,
+    fanInserted = 40,
+    fanRedundancyLost = 41,
+    fanRedundancyRegained = 42,
+    fanRemoved = 43,
+    firmwareActivationCompleted = 44,
+    firmwareActivationFailed = 45,
+    firmwareActivationStarted = 46,
+    firmwareResiliencyError = 47,
+    firmwareUpdateCompleted = 48,
+    firmwareUpdateFailed = 49,
+    firmwareUpdateStaged = 50,
+    firmwareUpdateStarted = 51,
+    generalFirmwareSecurityViolation = 52,
+    iPMIWatchdog = 53,
+    intelUPILinkWidthReducedToHalf = 54,
+    intelUPILinkWidthReducedToQuarter = 55,
+    invalidLoginAttempted = 56,
+    invalidUpload = 57,
+    inventoryAdded = 58,
+    inventoryRemoved = 59,
+    lanLost = 60,
+    lanRegained = 61,
+    legacyPCIPERR = 62,
+    legacyPCISERR = 63,
+    mEAutoConfigFailed = 64,
+    mEAutoConfigSuccess = 65,
+    mEBootGuardHealthEvent = 66,
+    mECpuDebugCapabilityDisabled = 67,
+    mEDirectFlashUpdateRequested = 68,
+    mEExceptionDuringShutdown = 69,
+    mEFactoryResetError = 70,
+    mEFactoryRestore = 71,
+    mEFirmwareException = 72,
+    mEFirmwarePanicReason = 73,
+    mEFirmwareRecoveryReason = 74,
+    mEFirmwareResiliencyError = 75,
+    mEFlashEraseError = 76,
+    mEFlashStateInformation = 77,
+    mEFlashStateInformationWritingEnabled = 78,
+    mEFlashVerificationError = 79,
+    mEFlashWearOutWarning = 80,
+    mEImageExecutionFailed = 81,
+    mEInternalError = 82,
+    mEManufacturingError = 83,
+    mEMctpInterfaceError = 84,
+    mEMultiPchModeMisconfig = 85,
+    mEPeciOverDmiError = 86,
+    mEPttHealthEvent = 87,
+    mERecoveryGpioForced = 88,
+    mERestrictedMode = 89,
+    mESmbusLinkFailure = 90,
+    mEUmaError = 91,
+    mEUnsupportedFeature = 92,
+    manufacturingModeEntered = 93,
+    manufacturingModeExited = 94,
+    memoryECCCorrectable = 95,
+    memoryECCUncorrectable = 96,
+    memoryExtendedECCCEData = 97,
+    memoryExtendedECCUEData = 98,
+    memoryParityCommandAndAddress = 99,
+    memoryParityNotKnown = 100,
+    memoryRASConfigurationDisabled = 101,
+    memoryRASConfigurationEnabled = 102,
+    memoryRASModeDisabled = 103,
+    memoryRASModeEnabled = 104,
+    memoryThermTrip = 105,
+    mirroringRedundancyDegraded = 106,
+    mirroringRedundancyFull = 107,
+    nMIButtonPressed = 108,
+    nMIDiagnosticInterrupt = 109,
+    pCIeCorrectableAdvisoryNonFatal = 110,
+    pCIeCorrectableBadDLLP = 111,
+    pCIeCorrectableBadTLP = 112,
+    pCIeCorrectableHeaderLogOverflow = 113,
+    pCIeCorrectableInternal = 114,
+    pCIeCorrectableLinkBWChanged = 115,
+    pCIeCorrectableReceiverError = 116,
+    pCIeCorrectableReplayNumRollover = 117,
+    pCIeCorrectableReplayTimerTimeout = 118,
+    pCIeCorrectableUnspecifiedAERError = 119,
+    pCIeFatalACSViolation = 120,
+    pCIeFatalAtomicEgressBlocked = 121,
+    pCIeFatalCompleterAbort = 122,
+    pCIeFatalCompletionTimeout = 123,
+    pCIeFatalDataLinkLayerProtocol = 124,
+    pCIeFatalECRCError = 125,
+    pCIeFatalFlowControlProtocol = 126,
+    pCIeFatalMCBlockedTLP = 127,
+    pCIeFatalMalformedTLP = 128,
+    pCIeFatalPoisonedTLP = 129,
+    pCIeFatalReceivedErrNonFatalMessage = 130,
+    pCIeFatalReceivedFatalMessageFromDownstream = 131,
+    pCIeFatalReceiverBufferOverflow = 132,
+    pCIeFatalSurpriseLinkDown = 133,
+    pCIeFatalTLPPrefixBlocked = 134,
+    pCIeFatalUncorrectableInternal = 135,
+    pCIeFatalUnexpectedCompletion = 136,
+    pCIeFatalUnspecifiedNonAERFatalError = 137,
+    pCIeFatalUnsupportedRequest = 138,
+    powerButtonPressed = 139,
+    powerButtonLongPressed = 140,
+    powerRestorePolicyApplied = 141,
+    powerSupplyConfigurationError = 142,
+    powerSupplyConfigurationErrorRecovered = 143,
+    powerSupplyFailed = 144,
+    powerSupplyFailurePredicted = 145,
+    powerSupplyFanFailed = 146,
+    powerSupplyFanRecovered = 147,
+    powerSupplyInserted = 148,
+    powerSupplyPowerGoodFailed = 149,
+    powerSupplyPowerLost = 150,
+    powerSupplyPowerRestored = 151,
+    powerSupplyPredictedFailureRecovered = 152,
+    powerSupplyRecovered = 153,
+    powerSupplyRemoved = 154,
+    powerUnitDegradedFromNonRedundant = 155,
+    powerUnitDegradedFromRedundant = 156,
+    powerUnitNonRedundantFromInsufficient = 157,
+    powerUnitNonRedundantInsufficient = 158,
+    powerUnitNonRedundantSufficient = 159,
+    powerUnitRedundancyDegraded = 160,
+    powerUnitRedundancyLost = 161,
+    powerUnitRedundancyRegained = 162,
+    resetButtonPressed = 163,
+    sELEntryAdded = 164,
+    securityBoot2ndFlashEnabled = 165,
+    securityP2aBridgeEnabled = 166,
+    securityUartPortDebugEnabled = 167,
+    securityUserNonRootUidZeroAssigned = 168,
+    securityUserNonRootUidZeroRemoved = 169,
+    securityUserRootDisabled = 170,
+    securityUserRootEnabled = 171,
+    securityUserStrongHashAlgoRestored = 172,
+    securityUserUnsupportedShellEnabled = 173,
+    securityUserUnsupportedShellRemoved = 174,
+    securityUserWeakHashAlgoEnabled = 175,
+    sensorThresholdCriticalHighGoingHigh = 176,
+    sensorThresholdCriticalHighGoingLow = 177,
+    sensorThresholdCriticalLowGoingHigh = 178,
+    sensorThresholdCriticalLowGoingLow = 179,
+    sensorThresholdWarningHighGoingHigh = 180,
+    sensorThresholdWarningHighGoingLow = 181,
+    sensorThresholdWarningLowGoingHigh = 182,
+    sensorThresholdWarningLowGoingLow = 183,
+    serviceFailure = 184,
+    serviceStarted = 185,
+    sparingRedundancyDegraded = 186,
+    sparingRedundancyFull = 187,
+    ssbThermalTrip = 188,
+    systemInterfaceDisabledProvisioned = 189,
+    systemInterfaceUnprovisioned = 190,
+    systemInterfaceWhitelistProvisioned = 191,
+    systemPowerGoodFailed = 192,
+    systemPowerLost = 193,
+    systemPowerOffFailed = 194,
+    systemPowerOnFailed = 195,
+    totalPowerConsumptionDropBelowTheLimit = 196,
+    totalPowerConsumptionExceedTheLimit = 197,
+    voltageRegulatorOverheated = 198,
 };
 } // namespace redfish::registries::openbmc
-- 
2.43.0

